application/server/server.go

```diff
@@ -183,8 +183,13 @@ func workspaceDidChangeWorkspaceFoldersHandler(srv *jrpc2.Server) jrpc2.Handler
 
 		logger.Info().Msg("RECEIVING")
 		defer logger.Info().Msg("SENDING")
-		workspace.Get().ChangeWorkspaceFolders(bgCtx, params)
-		command.HandleFolders(bgCtx, srv, di.Notifier())
+		changedFolders := workspace.Get().ChangeWorkspaceFolders(params)
+		command.HandleFolders(bgCtx, srv, di.Notifier(), di.ScanPersister())
```

Comment:
i moved the persister init logic to the folder handlers
This refactoring here was to make sure that the persister is initialized before a new f.ScanFolder is triggered for the new added folder